### PR TITLE
Clean remaining non_joined code and tests

### LINKED ISSUE #7308: Clean non-joined stream related comments and code
## Enhancement
In https://github.com/pingcap/tiflash/pull/7267, renamed NonJoinedBlockInputStream to ScanHashMapAfterProbeBlockInputStream, many remain code and comments need to be changed also.

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #7308 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

File: dbms/src/Interpreters/ExpressionActions.h [hunk 1]
```diff
@@ -201,7 +201,7 @@ class ExpressionActions
 
     static std::string getSmallestColumn(const NamesAndTypesList & columns);
 
-    BlockInputStreamPtr createStreamWithNonJoinedDataIfFullOrRightJoin(const Block & source_header, size_t index, size_t step, size_t max_block_size) const;
+    BlockInputStreamPtr createStreamWithScanHashMapAfterProbeIfNeeded(const Block & source_header, size_t index, size_t step, size_t max_block_size) const;
```
Comment (SeaRise) [inline dbms/src/Interpreters/ExpressionActions.h:204]: Seems that this function is not used in this pr.

**Review by SeaRise — APPROVED** [review]
Others LGTM

File: dbms/src/Interpreters/ExpressionActions.h (hunk 1 above)
Comment (yibin87) [reply dbms/src/Interpreters/ExpressionActions.h:204]: Yeah, the only usage of this function has been removed also. Will remove this useless function.

**Pull request merged**